starlite/contrib/jwt/jwt_auth.py

```diff
@@ -0,0 +1,359 @@
+from datetime import datetime, timedelta, timezone
+from typing import Any, Dict, Generic, Literal, Optional, Type, Union
+
+from pydantic_openapi_schema.v3_1_0 import (
+    Components,
+    OAuthFlow,
+    OAuthFlows,
+    SecurityRequirement,
+    SecurityScheme,
+)
+
+from starlite import Cookie, DefineMiddleware, Response
+from starlite.contrib.jwt.jwt_token import Token
+from starlite.contrib.jwt.middleware import (
+    JWTAuthenticationMiddleware,
+    JWTCookieAuthenticationMiddleware,
+)
+from starlite.enums import MediaType
+from starlite.security.base import AbstractSecurityConfig, UserType
+from starlite.status_codes import HTTP_201_CREATED
+
+
+class JWTAuth(Generic[UserType], AbstractSecurityConfig[UserType, Token]):
+    """JWT Authentication Configuration.
+
+    This class is the main entry point to the library, and it includes methods to create the middleware, provide login
+    functionality, and create OpenAPI documentation.
+    """
+
+    algorithm: str = "HS256"
```

Comment:
I think this could be a `Literal`, just for convenience.

Reply:
no, there are too many options.

Reply:
Too bad